README.md
<img align="center" src="https://raw.githubusercontent.com/arcticicestudio/northem-java/develop/src/main/assets/media/banner.png" alt="Northem Java"/> <a href="https://github.com/arcticicestudio/northem"><img align="right" src="https://raw.githubusercontent.com/arcticicestudio/northem-java/develop/src/main/assets/media/northem-logo.png"/></a>

<img src="https://cdn.travis-ci.org/images/favicon-c566132d45ab1a9bcae64d8d90e4378a.svg" width=24 height=24/> [![Travis CI](https://img.shields.io/travis/arcticicestudio/northem-java/develop.svg)](https://travis-ci.org/arcticicestudio/northem-java) <img src="https://codecov.io/favicon.ico" width=24 height=24/> [![codecov](https://codecov.io/gh/arcticicestudio/northem-java/branch/develop/graph/badge.svg)](https://codecov.io/gh/arcticicestudio/northem-java) <img src="https://assets-cdn.github.com/favicon.ico" width=24 height=24/> [![GitHub release](https://img.shields.io/github/release/arcticicestudio/northem-java.svg)](https://github.com/arcticicestudio/northem-java/releases/latest) <img src="http://central.sonatype.org/favicon.ico" width=24 height=24/> [![Central Repository](https://img.shields.io/maven-central/v/com.arcticicestudio/northem-java.svg)](http://search.maven.org/#search%7Cgav%7C1%7Cg%3A%22com.arcticicestudio%22%20AND%20a%3A%22northem-java%22) <img src="https://oss.sonatype.org/favicon.ico"/> [![OSS Sonatype Snapshot](https://img.shields.io/badge/snapshot-0.3.0--SNAPSHOT-blue.svg)](https://oss.sonatype.org/content/repositories/snapshots/com/arcticicestudio/northem-java)

Java library implementation of the north-bluish dark color palette [Northem](https://github.com/arcticicestudio/northem).  
Optimized for syntax highlighting- and UI projects.  

## Getting started
### Setup
To use northem-java it must be available in your classpath.  
You can get it from the <img src="http://central.sonatype.org/favicon.ico" width=16 height=16/> <a href="https://search.maven.org">Central Repository</a> as a dependency for your favorite build tool or [download the latest version](https://github.com/arcticicestudio/northem-java/releases/latest).

<img src="http://apache.org/favicons/favicon.ico" width=16 height=16/> <a href="https://maven.apache.org">Apache Maven</a>
```xml
<dependency>
  <groupId>com.arcticicestudio</groupId>
  <artifactId>northem-java</artifactId>
  <version>0.2.0</version>
</dependency>
```

<img src="https://gradle.org/wp-content/uploads/fbrfg/favicon.ico" width=16 height=16/> <a href="https://gradle.org">Gradle</a>
```java
compile(group: 'com.arcticicestudio', name: 'northem-java', version: '0.2.0')
```

<img src="http://apache.org/favicons/favicon.ico" width=16 height=16/> <a href="https://ant.apache.org/ivy">Apache Ivy</a>
```xml
<dependency org="com.arcticicestudio" name="northem-java" rev="0.2.0" />
```

Development snapshots are available via [OSS Sonatype](https://oss.sonatype.org/content/repositories/snapshots/com/arcticicestudio/northem-java).  

### Build
Build and install northem-java into your local repository without GPG signing:  
```
mvn clean install
```

Signed artifacts may be build by using the `sign-gpg` profile with a provided `gpg.keyname` property:
```
mvn clean install -Dgpg.keyname=YourGPGKeyId
```

A continuous integration build is running at [Travis-CI](https://travis-ci.org/arcticicestudio/northem-java).

## Variations
  - <a href="https://github.com/arcticicestudio/northem-dark-java"><img src="https://java.com/favicon.ico" width=16 height=16> Northem Dark - Java</a> <img src="https://www.kernel.org/theme/images/logos/favicon.png" width=16 height=16 /> <img src="https://developer.apple.com/favicon.ico" width=16 height=16 /> <img src="https://www.microsoft.com/en-us/windows/favicon.ico" width=16 height=16 />
  - <a href="https://github.com/arcticicestudio/northem-light-java"><img src="https://java.com/favicon.ico" width=16 height=16> Northem Light - Java</a> <img src="https://www.kernel.org/theme/images/logos/favicon.png" width=16 height=16 /> <img src="https://developer.apple.com/favicon.ico" width=16 height=16 /> <img src="https://www.microsoft.com/en-us/windows/favicon.ico" width=16 height=16 />

## Other Projects
  - <a href="https://github.com/arcticicestudio/northem-intellij-idea-syntax"><img src="https://www.jetbrains.com/_assets//shared/favicons/jetbrains.ico" width=16 height=16> Northem IntelliJ IDEA Syntax</a> <img src="https://www.kernel.org/theme/images/logos/favicon.png" width=16 height=16 /> <img src="https://developer.apple.com/favicon.ico" width=16 height=16 /> <img src="https://www.microsoft.com/en-us/windows/favicon.ico" width=16 height=16 />
  - <a href="https://github.com/arcticicestudio/northem-atom-syntax"><img src="https://atom.io/favicon.ico" width=16 height=16> Northem Atom Syntax</a> <img src="https://www.kernel.org/theme/images/logos/favicon.png" width=16 height=16 /> <img src="https://developer.apple.com/favicon.ico" width=16 height=16 /> <img src="https://www.microsoft.com/en-us/windows/favicon.ico" width=16 height=16 />
  - <a href="https://github.com/arcticicestudio/northem-atom-ui"><img src="https://atom.io/favicon.ico" width=16 height=16> Northem Atom UI</a> <img src="https://www.kernel.org/theme/images/logos/favicon.png" width=16 height=16 /> <img src="https://developer.apple.com/favicon.ico" width=16 height=16 /> <img src="https://www.microsoft.com/en-us/windows/favicon.ico" width=16 height=16 />
  - <a href="https://github.com/arcticicestudio/northem-eclipse-syntax"><img src="https://eclipse.org/favicon.ico" width=16 height=16> Northem Eclipse Syntax</a> <img src="https://www.kernel.org/theme/images/logos/favicon.png" width=16 height=16 /> <img src="https://developer.apple.com/favicon.ico" width=16 height=16 /> <img src="https://www.microsoft.com/en-us/windows/favicon.ico" width=16 height=16 />
  - <a href="https://github.com/arcticicestudio/northem-gedit"><img src="https://static.gnome.org/wiki.gnome.org/gnome/css/favicon.png" width=16 height=16> Northem gedit</a> <img src="https://www.kernel.org/theme/images/logos/favicon.png" width=16 height=16 />
  - <a href="#"><img src="http://www.gtk.org/images/gtk-logo.ico" width=16 height=16> Northem GTK (in drafting)</a> <img src="https://www.kernel.org/theme/images/logos/favicon.png" width=16 height=16 />
  - <a href="https://github.com/arcticicestudio/northem-notepadplusplus"><img src="http://notepad-plus-plus.org/assets/images/favicon.ico" width=16 height=16> Northem Notepad++</a> <img src="https://www.microsoft.com/en-us/windows/favicon.ico" width=16 height=16 />

Visit the [official website](http://arcticicestudio.com/northem) for more information.

## Development
[![](https://img.shields.io/badge/Changelog-v0.2.0-blue.svg)](https://github.com/arcticicestudio/northem-java/blob/v0.2.0/CHANGELOG.md)

### Workflow
This project follows the [gitflow](http://nvie.com/posts/a-successful-git-branching-model) branching model.

### Specifications
This project follows the [Arctic Versioning Specification (ArcVer)](https://github.com/arcticicestudio/arcver).

### Contribution
Please report issues/bugs, suggestions for improvements and feature requests to the [issuetracker](https://github.com/arcticicestudio/northem-java/issues).

---

<img src="http://arcticicestudio.com/favicon.ico" width=16 height=16/> Copyright &copy; 2016 Arctic Ice Studio

[![GPL-3.0](http://www.gnu.org/graphics/gplv3-88x31.png)](http://www.gnu.org/licenses/gpl.txt) [![CC BY-SA-4.0](http://mirrors.creativecommons.org/presskit/buttons/88x31/svg/by-sa.svg)](http://creativecommons.org/licenses/by-sa/4.0/)


CHANGELOG.md
Northem - Java
==============

## 0.2.0 (2016-08-02) - Apache Maven Migration
This version is mainly focused on a [Apache Maven](https://maven.apache.org) migration.
Release and snapshot versions are now deployed to the [Central Repository](https://search.maven.org) via [OSS Sonatype](https://oss.sonatype.org).

### Improvements
#### Build Tool
  - Migrated to [Apache Maven](https://maven.apache.org) based on the [`glacier-apache-maven@0.3.0`](https://github.com/arcticicestudio/glacier-apache-maven) project skeletons

#### Documentation
  - The `README` has been restyled and now includes a "Getting started" section

## 0.1.0 (2016-04-09)
### Features
  - Implemented the [Northem](https://github.com/arcticicestudio/northem) color palette enumeration class [`Northem`](https://github.com/arcticicestudio/northem-java/blob/master/src/main/java/com/arcticicestudio/northem/Northem.java)  

**Constants**

| Name           | RGB Value     |
| -------------- | ------------- |
| `DARKEST_GRAY` | `45,45,46`    |
| `DARK_GRAY`    | `51,51,52`    |
| `GRAY`         | `56,56,57`    |
| `LIGHT_GRAY`   | `96,96,98`    |
| `DIRTY_WHITE`  | `222,222,222` |
| `WHITE`        | `236,238,239` |
| `AQUA`         | `144,205,199` |
| `LIGHT_BLUE`   | `134,193,210` |
| `BLUE`         | `136,178,205` |
| `DARK_BLUE`    | `92,129,174`  |
| `RED`          | `223,117,116` |
| `YELLOW`       | `252,206,133` |
| `ORANGE`       | `239,160,112` |
| `LIME`         | `200,203,121` |
| `PURPLE`       | `194,162,201` |

**Methods**

| Name | Description |
| ---- | ----------- |
| `Northem(final int RED, final int GREEN, final int BLUE)` | Constructs a new color object. |
| `public static String rgb(final Color COLOR)` | Converts the color to the RGB identifier. |
| `public static String hex(final Color COLOR)` | Converts the color to the HEX identifier. |
| `public Color get()` | Returns the color object. |

## 0.0.0 (2016-04-09) - Repository Reinitialization


src/test/java/com/arcticicestudio/northem/NorthemTest.java
/*
+++++++++++++++++++++++++++++++++++++++++++
title     Northem Enumeration JUnit Test  +
project   northem-java                    +
file      NorthemTest.java                +
version   0.1.0                           +
author    Arctic Ice Studio               +
email     development@arcticicestudio.com +
website   http://arcticicestudio.com      +
copyright Copyright (C) 2016              +
created   2016-01-23 10:24 UTC+0100       +
modified  2016-04-09 12:20 UTC+0200       +
+++++++++++++++++++++++++++++++++++++++++++

[Description]
JUnit test class for the 'Northem' class of the "Northem - Java" API.

[Copyright]
Copyright (C) 2016 Arctic Ice Studio <development@arcticicestudio.com>

[References]
Northem
  (https://github.com/arcticicestudio/northem)
Java 8 API Documentation
  (https://docs.oracle.com/javase/8/docs/api/)
JavaFX 2.2 API Documentation
  (https://docs.oracle.com/javafx/2/api/)
JavaFX 2.2 Oracle Project Website
  (http://docs.oracle.com/javase/8/javase-clienttechnologies.htm)
Arctic Versioning Specification
  (http://specs.arcticicestudio.com/arcver) (ArcVer)
*/

package com.arcticicestudio.northem;

import org.junit.Test;
import static org.junit.Assert.assertEquals;

/**
 * JUnit test class for the {@link Northem} class of the
 * <a href="https://github.com/arcticicestudio/northem-java">Northem - Java</a> API.
 *
 * @author Arctic Ice Studio &lt;development@arcticicestudio.com&gt;
 * @see Northem
 * @see <a href="http://junit.org/">http://junit.org/</a>
 * @see <a href="https://github.com/arcticicestudio/northem-java">Northem - Java</a>
 * @since 0.1.0
 */
public class NorthemTest {
  @Test
  public void shouldReturnTheCorrectHEXColorCode() {
    assertEquals(Northem.hex(Northem.DARKEST_GRAY.get()), "#2D2D2E");
    assertEquals(Northem.hex(Northem.DARK_GRAY.get()), "#333334");
    assertEquals(Northem.hex(Northem.GRAY.get()), "#383839");
    assertEquals(Northem.hex(Northem.LIGHT_GRAY.get()), "#606062");
    assertEquals(Northem.hex(Northem.DIRTY_WHITE.get()), "#DEDEDE");
    assertEquals(Northem.hex(Northem.WHITE.get()), "#ECEEEF");
    assertEquals(Northem.hex(Northem.AQUA.get()), "#90CDC7");
    assertEquals(Northem.hex(Northem.LIGHT_BLUE.get()), "#86C1D2");
    assertEquals(Northem.hex(Northem.BLUE.get()), "#88B2CD");
    assertEquals(Northem.hex(Northem.DARK_BLUE.get()), "#5C81AE");
    assertEquals(Northem.hex(Northem.RED.get()), "#DF7574");
    assertEquals(Northem.hex(Northem.YELLOW.get()), "#FCCE85");
    assertEquals(Northem.hex(Northem.ORANGE.get()), "#EFA070");
    assertEquals(Northem.hex(Northem.LIME.get()), "#C8CB79");
    assertEquals(Northem.hex(Northem.PURPLE.get()), "#C2A2C9");
  }

  @Test
  public void shouldReturnTheCorrectRGBColorCode() {
    assertEquals(Northem.rgb(Northem.DARKEST_GRAY.get()), "rgb(45, 45, 46)");
    assertEquals(Northem.rgb(Northem.DARK_GRAY.get()), "rgb(51, 51, 52)");
    assertEquals(Northem.rgb(Northem.GRAY.get()), "rgb(56, 56, 57)");
    assertEquals(Northem.rgb(Northem.LIGHT_GRAY.get()), "rgb(96, 96, 98)");
    assertEquals(Northem.rgb(Northem.DIRTY_WHITE.get()), "rgb(222, 222, 222)");
    assertEquals(Northem.rgb(Northem.WHITE.get()), "rgb(236, 238, 239)");
    assertEquals(Northem.rgb(Northem.AQUA.get()), "rgb(144, 205, 199)");
    assertEquals(Northem.rgb(Northem.LIGHT_BLUE.get()), "rgb(134, 193, 210)");
    assertEquals(Northem.rgb(Northem.BLUE.get()), "rgb(136, 178, 205)");
    assertEquals(Northem.rgb(Northem.DARK_BLUE.get()), "rgb(92, 129, 174)");
    assertEquals(Northem.rgb(Northem.RED.get()), "rgb(223, 117, 116)");
    assertEquals(Northem.rgb(Northem.YELLOW.get()), "rgb(252, 206, 133)");
    assertEquals(Northem.rgb(Northem.ORANGE.get()), "rgb(239, 160, 112)");
    assertEquals(Northem.rgb(Northem.LIME.get()), "rgb(200, 203, 121)");
    assertEquals(Northem.rgb(Northem.PURPLE.get()), "rgb(194, 162, 201)");
  }
}


src/main/java/com/arcticicestudio/northem/package-info.java
/*
+++++++++++++++++++++++++++++++++++++++++++
title     Package Info                    +
project   northem-java                    +
file      package-info.java               +
version   0.2.0                           +
author    Arctic Ice Studio               +
email     development@arcticicestudio.com +
website   http://arcticicestudio.com      +
copyright Copyright (C) 2016              +
created   2016-01-23 10:15 UTC+0100       +
modified  2016-04-09 12:02 UTC+0200       +
+++++++++++++++++++++++++++++++++++++++++++

[Description]
Root package of the "Northem - Java" project, which implements the "Northem" color palette as enumeration.
The package represents the public API.
The main entry point of the package is the "Northem" class.

[Copyright]
Copyright (C) 2016 Arctic Ice Studio <development@arcticicestudio.com>

[References]
Northem
  (https://github.com/arcticicestudio/northem)
Java 8 API Documentation
  (https://docs.oracle.com/javase/8/docs/api/)
JavaFX 2.2 API Documentation
  (https://docs.oracle.com/javafx/2/api/)
JavaFX 2.2 Oracle Project Website
  (http://docs.oracle.com/javase/8/javase-clienttechnologies.htm)
Arctic Versioning Specification
  (http://specs.arcticicestudio.com/arcver) (ArcVer)
*/
/**
 * Root package of the <a href="https://github.com/arcticicestudio/northem-java">Northem - Java</a>
 * project, which implements the <a href="https://github.com/arcticicestudio/northem">Northem</a>
 * color palette as enumeration.
 * The package represents the public API.
 * The main entry point of the package is the {@link com.arcticicestudio.northem.Northem} class.
 *
 * @author Arctic Ice Studio &lt;development@arcticicestudio.com&gt;
 * @see <a href="https://github.com/arcticicestudio/northem-java">Northem - Java</a>
 * @see <a href="https://github.com/arcticicestudio/northem">Northem</a>
 * @version 0.2.0
 * @since 0.1.0
 */
package com.arcticicestudio.northem;


src/main/java/com/arcticicestudio/northem/Northem.java
/*
+++++++++++++++++++++++++++++++++++++++++++
title     Northem                         +
project   northem-java                    +
file      Northem.java                    +
version   0.2.0                           +
author    Arctic Ice Studio               +
email     development@arcticicestudio.com +
website   http://arcticicestudio.com      +
copyright Copyright (C) 2016              +
created   2016-01-20 14:25 UTC+0100       +
modified  2016-04-09 12:17 UTC+0200       +
+++++++++++++++++++++++++++++++++++++++++++

[Description]
"Northem" color palette as enumeration for the "Northem - Java" project.

[Copyright]
Copyright (C) 2016 Arctic Ice Studio <development@arcticicestudio.com>

[References]
Northem
  (https://github.com/arcticicestudio/northem)
Java 8 API Documentation
  (https://docs.oracle.com/javase/8/docs/api/)
JavaFX 2.2 API Documentation
  (https://docs.oracle.com/javafx/2/api/)
JavaFX 2.2 Oracle Project Website
  (http://docs.oracle.com/javase/8/javase-clienttechnologies.htm)
Arctic Versioning Specification
  (http://specs.arcticicestudio.com/arcver) (ArcVer)
*/
package com.arcticicestudio.northem;

import javafx.scene.paint.Color;
import java.util.regex.Pattern;

/**
 * <a href="https://github.com/arcticicestudio/northem">Northem</a> color palette
  * as enumeration of the <a href="https://github.com/arcticicestudio/northem-java">Northem - Java</a> project.
 * <div>
 *   <table style="display:inline-block">
 *     <thead>
 *       <th style="border:none;width:25px;height:30px;background-color:rgb(45,45,46);margin: 0"></th>
 *       <th style="border:none;width:25px;height:30px;background-color:rgb(51,51,52);margin: 0"></th>
 *       <th style="border:none;width:25px;height:30px;background-color:rgb(56,56,57);margin: 0"></th>
 *       <th style="border:none;width:25px;height:30px;background-color:rgb(96,96,98);margin: 0"></th>
 *       <th style="border:none;width:25px;height:30px;background-color:rgb(222,222,222);margin: 0"></th>
 *       <th style="border:none;width:25px;height:30px;background-color:rgb(236,238,239);margin: 0"></th>
 *       <th style="border:none;width:25px;height:30px;background-color:rgb(144,205,199);margin: 0"></th>
 *       <th style="border:none;width:25px;height:30px;background-color:rgb(134,193,210);margin: 0"></th>
 *       <th style="border:none;width:25px;height:30px;background-color:rgb(136,178,205);margin: 0"></th>
 *       <th style="border:none;width:25px;height:30px;background-color:rgb(92,129,174);margin: 0"></th>
 *       <th style="border:none;width:25px;height:30px;background-color:rgb(223,117,116);margin: 0"></th>
 *       <th style="border:none;width:25px;height:30px;background-color:rgb(252,206,133);margin: 0"></th>
 *       <th style="border:none;width:25px;height:30px;background-color:rgb(239,160,112);margin: 0"></th>
 *       <th style="border:none;width:25px;height:30px;background-color:rgb(200,203,121);margin: 0"></th>
 *       <th style="border:none;width:25px;height:30px;background-color:rgb(194,162,201);margin: 0"></th>
 *     </thead>
 *   </table>
 * </div>
 *
 * @author Arctic Ice Studio &lt;development@arcticicestudio.com&gt;
 * @see <a href="https://github.com/arcticicestudio/northem">"Northem" GitHub Repository</a>
 * @see <a href="https://github.com/arcticicestudio/northem-java">"Northem - Java" GitHub Repository</a>
 * @since 0.1.0
 */
public enum Northem {
  /**
   * The color {@code darkest-gray} with an RGB value of {@code rgb(45,45,46)} and HEX value of {@code #2D2D2E}.
   * <div style="border:none;width:25px;height:30px;background-color:rgb(45,45,46);margin: 0"></div>
   */
  DARKEST_GRAY(45,45,46),
  /**
   * The color {@code dark-gray} with an RGB value of {@code rgb(51,51,52)} and HEX value of {@code #333334}.
   * <div style="border:none;width:25px;height:30px;background-color:rgb(51,51,52);margin: 0"></div>
   */
  DARK_GRAY(51,51,52),
  /**
   * The color {@code gray} with an RGB value of {@code rgb(56,56,57)} and HEX value of {@code #383839}.
   * <div style="border:none;width:25px;height:30px;background-color:rgb(56,56,57);margin: 0"></div>
   */
  GRAY(56,56,57),
  /**
   * The color {@code light-gray} with an RGB value of {@code rgb(96,96,98)} and HEX value of {@code #606062}.
   * <div style="border:none;width:25px;height:30px;background-color:rgb(96,96,98);margin: 0"></div>
   */
  LIGHT_GRAY(96,96,98),
  /**
   * The color {@code dirty-white} with an RGB value of {@code rgb(222,222,222)} and HEX value of {@code #DEDEDE}.
   * <div style="border:none;width:25px;height:30px;background-color:rgb(222,222,222);margin: 0"></div>
   */
  DIRTY_WHITE(222,222,222),
  /**
   * The color {@code white} with an RGB value of {@code rgb(236,238,239)} and HEX value of {@code #ECEEEF}.
   * <div style="border:none;width:25px;height:30px;background-color:rgb(236,238,239);margin: 0"></div>
   */
  WHITE(236,238,239),
  /**
   * The color {@code aqua} with an RGB value of {@code rgb(144,205,199)} and HEX value of {@code #90CDC7}.
   * <div style="border:none;width:25px;height:30px;background-color:rgb(144,205,199);margin: 0"></div>
   */
  AQUA(144,205,199),
  /**
   * The color {@code light-blue} with an RGB value of {@code rgb(134,193,210)} and HEX value of {@code #86C1D2}.
   * <div style="border:none;width:25px;height:30px;background-color:rgb(134,193,210);margin: 0"></div>
   */
  LIGHT_BLUE(134,193,210),
  /**
   * The color {@code blue} with an RGB value of {@code rgb(136,178,205)} and HEX value of {@code #88B2CD}.
   * <div style="border:none;width:25px;height:30px;background-color:rgb(136,178,205);margin: 0"></div>
   */
  BLUE(136,178,205),
  /**
   * The color {@code dark-blue} with an RGB value of {@code rgb(92,129,174)} and HEX value of {@code #5C81AE}.
   * <div style="border:none;width:25px;height:30px;background-color:rgb(92,129,174);margin: 0"></div>
   */
  DARK_BLUE(92,129,174),
  /**
   * The color {@code red} with an RGB value of {@code rgb(223,117,116)} and HEX value of {@code #DF7574}.
   * <div style="border:none;width:25px;height:30px;background-color:rgb(223,117,116);margin: 0"></div>
   */
  RED(223,117,116),
  /**
   * The color {@code yellow} with an RGB value of {@code rgb(252,206,133)} and HEX value of {@code #FCCE85}.
   * <div style="border:none;width:25px;height:30px;background-color:rgb(252,206,133);margin: 0"></div>
   */
  YELLOW(252,206,133),
  /**
   * The color {@code orange} with an RGB value of {@code rgb(239,160,112)} and HEX value of {@code #EFA070}.
   * <div style="border:none;width:25px;height:30px;background-color:rgb(239,160,112);margin: 0"></div>
   */
  ORANGE(239,160,112),
  /**
   * The color {@code lime} with an RGB value of {@code rgb(200,203,121)} and HEX value of {@code #C8CB79}.
   * <div style="border:none;width:25px;height:30px;background-color:rgb(200,203,121);margin: 0"></div>
   */
  LIME(200,203,121),
  /**
   * The color {@code purple} with an RGB value of {@code rgb(194,162,201)} and HEX value of {@code #C2A2C9}.
   * <div style="border:none;width:25px;height:30px;background-color:rgb(194,162,201);margin: 0"></div>
   */
  PURPLE(194,162,201);

  /**
   * Compiled pattern to convert the RGB and HEX color values.
   */
  private static final Pattern COLOR_PREFIX = Pattern.compile("0x", Pattern.LITERAL);

  /**
   * Contains the RGB color code.
   */
  private final Color COLOR;

  /**
   * Constructs a new color object.
   *
   * @param RED The value for the additive primary color {@code red}
   * @param GREEN The value for the additive primary color {@code green}
   * @param BLUE The value for the additive primary color {@code blue}
   */
  Northem(final int RED, final int GREEN, final int BLUE) {
    COLOR = Color.rgb(RED, GREEN, BLUE);
  }

  /**
   * Converts the color to the {@code RGB} identifier.
   *
   * @param COLOR The color object
   * @return a string representation of the {@code RGB} identifier
   */
  public static String rgb(final Color COLOR) {
    String hex = COLOR_PREFIX.matcher(COLOR.toString()).replaceAll("");
    String hexRed = hex.substring(0, 2).toUpperCase();
    String hexGreen = hex.substring(2, 4).toUpperCase();
    String hexBlue = hex.substring(4, 6).toUpperCase();

    String intRed = Integer.toString(Integer.parseInt(hexRed, 16));
    String intGreen = Integer.toString(Integer.parseInt(hexGreen, 16));
    String intBlue = Integer.toString(Integer.parseInt(hexBlue, 16));

    return String.join("", "rgb(", intRed, ", ", intGreen, ", ", intBlue, ")");
  }

  /**
   * Converts the color to the {@code HEX} identifier.
   *
   * @param COLOR The color object
   * @return a string representation of the {@code HEX} identifier
   */
  public static String hex(final Color COLOR) {
    return COLOR_PREFIX.matcher(COLOR.toString()).replaceAll("#").substring(0, 7).toUpperCase();
  }

  /**
   * Returns the color object.
   *
   * @return the color object
   */
  public Color get() {
    return COLOR;
  }
}
